rs/execution_environment/src/history.rs

```diff
@@ -145,7 +145,12 @@ impl IngressHistoryWriterImpl {
 impl IngressHistoryWriter for IngressHistoryWriterImpl {
     type State = ReplicatedState;
 
-    fn set_status(&self, state: &mut Self::State, message_id: MessageId, status: IngressStatus) {
+    fn set_status(
+        &self,
+        state: &mut Self::State,
+        message_id: MessageId,
+        status: IngressStatus,
+    ) -> Arc<IngressStatus> {
```

Comment:
It is not obvious what this function returns, can you please add comment or maybe change the name of the function?

Reply:
This is a method defined by the `IngressHistoryWriter` trait. We don't usually duplicate doc comments between traits and their implementations, as they are likely to diverge over time.